internal/auth/jwtauth.go

```diff
@@ -105,14 +105,33 @@ var userSubjectContextKey struct{}
 
 // GetUserSubjectFromContext returns the user subject from the context, or nil
 func GetUserSubjectFromContext(ctx context.Context) string {
-	subject, ok := ctx.Value(userSubjectContextKey).(string)
+	token, ok := ctx.Value(userSubjectContextKey).(openid.Token)
 	if !ok {
+		fmt.Printf("***\n")
+		fmt.Printf("no token in coGntext\n")
+		fmt.Printf("***\n")
 		return ""
 	}
-	return subject
+	return token.Subject()
 }
 
-// WithUserSubjectContext stores the specified user subject in the context.
-func WithUserSubjectContext(ctx context.Context, subject string) context.Context {
+// GetUserClaimFromContext returns the specified claim from the user subject in
+// the context if found and of the correct type
+func GetUserClaimFromContext[T any](ctx context.Context, claim string) (T, bool) {
+	var ret T
+	token, ok := ctx.Value(userSubjectContextKey).(openid.Token)
+	if !ok {
+		return ret, false
+	}
+	data, ok := token.Get(claim)
+	if !ok {
+		return ret, false
+	}
+	ret, ok = data.(T)
+	return ret, ok
+}
+
+// WithAuthTokenContext stores the specified user subject in the context.
+func WithAuthTokenContext(ctx context.Context, subject openid.Token) context.Context {
```

Comment:
Since we now store the token and not just the subject, the godocs, parameter and function names should also change. For example, `userSubjectContextKey` should be `userTokenContextKey`

Reply:
I did change the function name, I think, except that we now have one "store" and two "loads":

* `GetUserSubjectFromContext` returns the `sub` claim
* `GetUserClaimFromContext` returns other claims, or `"", false` if not present.